Sources/CartonCLI/Commands/Init.swift

```diff
@@ -25,43 +25,85 @@ struct Init: ParsableCommand {
+    transform: { Templates(rawValue: $0.lowercased()) }
+  )
   var template: Templates?
 
-  @Option(name: .long,
-          help: "The name of the project") var name: String?
+  @Option(
+    name: .long,
+    help: """
+          The name of the project, if it's different from the current working directory's name.
+          If this option is set, carton will create the package for the project in a subdirectory by the given name.
+          Otherwise, the package will be created in the current working directory, with the package named after the directory's.
+          """
+  )
+  var name: String?
 
   func run() throws {
     let terminal = InteractiveWriter.stdout
 
-    guard let name = name ?? localFileSystem.currentWorkingDirectory?.basename else {
-      terminal.write("Project name could not be inferred\n", inColor: .red)
-      return
-    }
-    guard let currentDir = localFileSystem.currentWorkingDirectory else {
-      terminal.write("Failed to get current working directory.\n", inColor: .red)
-      return
+    guard let currentDirectory = localFileSystem.currentWorkingDirectory else {
+      throw ProjectInitialisationError.cannotFindCurrentDirectory
     }
+
+    /// The kind of template the project is to be created with.
+    /// - Note: This value is of the type `Templates`, not `Template`.
     let template = self.template ?? .basic
+
+    let projectName = name ?? currentDirectory.basename
+    let projectPath = name == nil ? currentDirectory : currentDirectory.appending(component: projectName)
+    
+    // FIXME: Replace with `NSString.abbreviatingWithTildeInPath`?
+    let packagePathRelativeToHomeDirectory = projectPath.relative(to: localFileSystem.homeDirectory)
+
     terminal.write("Creating new project with template ")
     terminal.write("\(template.rawValue)", inColor: .green)
     terminal.write(" in ")
-    terminal.write("\(name)\n", inColor: .cyan)
+    terminal.write("~/\(packagePathRelativeToHomeDirectory.pathString)\n", inColor: .cyan)
 
-    guard let packagePath = self.name == nil ?
-      localFileSystem.currentWorkingDirectory :
-      AbsolutePath(name, relativeTo: currentDir)
-    else {
-      terminal.write("Path to project could be created.\n", inColor: .red)
-      return
+    do {
+      // `LocalFileSystem.createDirectory` wraps `FileManager.createDirectory`, which throws an NSError.
+      try localFileSystem.createDirectory(projectPath)
+    } catch let nsError as NSError {
+        throw ProjectInitialisationError.cannotCreateDirectory(path: packagePathRelativeToHomeDirectory, reason: nsError.localizedDescription)
+    }
+    
+    // FIXME: Handle this error-catching in each `Template.create()` instead.
+    do {
+      // FIXME:  The first `template` is of type `Templates`, and two consecutive `template` looks odd.
+      // Maybe rename `Templates` as `TemplateKind`, and refactor `Template` so the template doesn't have to be created this way.
```

Comment:
I agree with the renaming, as I've mentioned in comments in the respective file. Feel free to rename in this PR or a separate one if you prefer that.